### PR TITLE
feat(apps): Deploy to an environment

### PR DESCRIPTION
**Pull request by raulb**
## Description of change

Fixes https://github.com/meroxa/cli/issues/595

## Type of change

- [x]  New feature
- [ ]  Bug fix
- [ ]  Refactor
- [ ]  Documentation

## How was this tested?

- [x]  Unit Tests
- [ ]  Tested in staging
- [ ]  Tested in minikube

## Demo

**Before**

```
$ meroxa apps deploy --spec 0.2.0
...
```

**After**

```
$ meroxa apps deploy --spec 0.2.0 --env my-env
...
```

### CODE REVIEW

**Review by anna-cross — APPROVED** [review]
Looks good to me!

File: cmd/meroxa/root/apps/deploy.go [hunk 1]
```diff
@@ -882,16 +884,20 @@ func (d *Deploy) Execute(ctx context.Context) error {
 	if d.specVersion == "" && d.lang == turbine.Ruby {
 		d.specVersion = ir.LatestSpecVersion
 	}
+
+	// Intermediate Representation Workflow (if using --spec on apps deploy)
```
[the remark this answers is not in the document]
Comment (raulb) [reply cmd/meroxa/root/apps/deploy.go:902]: @dianadoherty Please take a look! https://github.com/meroxa/cli/pull/605/commits/1652331f4e47b5781d38ef90b28081f16725c071

File: cmd/meroxa/root/apps/deploy.go [hunk 2]
```diff
@@ -834,8 +835,22 @@ func (d *Deploy) tearDownExistingResources(ctx context.Context) error {
 	return nil
 }
 
+// validateFlags take care of some possible incompatibilities between flag values.
+func (d *Deploy) validateFlags() error {
+	if d.flags.Spec == "" && d.flags.Environment != "" {
+		return fmt.Errorf(
+			"please run `meroxa apps deploy` with `--spec %s` or `--spec %s` if you want to deploy to an environment",
+			ir.SpecVersion_0_1_1, ir.SpecVersion_0_2_0)
```
Comment (dianadoherty) [inline cmd/meroxa/root/apps/deploy.go:843]: So this isn't true if the language is ruby, since it automatically uses a spec-ed version

Comment (raulb) [reply cmd/meroxa/root/apps/deploy.go:843]: @dianadoherty oops!

Comment (raulb) [reply cmd/meroxa/root/apps/deploy.go:845]: @dianadoherty it should be fixed now, thank you for the call out.

**Review by dianadoherty — APPROVED** [review]

**Pull request merged**